File: futurejava/core/src/main/java/org/javafish/board/Side.java
package org.javafish.board;

public class Side {
    public final static int WHITE = 0;
    public final static int BLACK = 1;

    public static int flip(int side){
        return BLACK ^ side;
    }

    public static int multiplicator(int side) {
        return side == WHITE ? 1 : -1;
    }
}

File: ts-engine/src/Side.ts
export const Side = {
  WHITE: 0,
  BLACK: 1,
} as const;

export type SideType = (typeof Side)[keyof typeof Side];

export namespace SideUtils {
  export function flip(side: SideType): SideType {
    return side === Side.BLACK ? Side.WHITE : Side.BLACK;
  }

  export function multiplicator(side: number): number {
    return side === Side.WHITE ? 1 : -1;
  }
}
